src/test/java/uk/ac/ebi/ampt2d/metadata/MetadataApplicationTest.java

```diff
@@ -305,6 +305,22 @@ private String postTestWebResource() throws Exception {
         return mvcResult.getResponse().getHeader("Location");
     }
 
+    @Test
+    public void postWebResourceInvalidURL() throws Exception {
+        WebResource testWebResource = new WebResource(WebResource.Type.CENTER_WEB, "");
+
+        mockMvc.perform(post("/webResources")
+                .content(testWebResourceJson.write(testWebResource).getJson()))
+                .andExpect(status().is4xxClientError());
+
+        testWebResource = new WebResource(WebResource.Type.CENTER_WEB, "https://www.ebi.ac.uk<>");
```

Comment:
Could you please add some more tests to check valid and invalid URLs? For instance, making sure that only HTTP, HTTPS and FTP protocols are considered valid, as mentioned during @Zhicheng-Liu's review. Testing arbitrary errors in the URL would be also a good addition.